docker-compose.yml

```diff
@@ -2,41 +2,52 @@ version: '3.8'
 
 services:
   nginx:
+    build:
+      context: .
+      dockerfile: sys/docker/alpine-nginx/Dockerfile
+      network: host
     container_name: badge-poser-nginx
-    image: nginx:alpine
-    working_dir: /application
     depends_on:
       - phpfpm
-    volumes:
-      - .:/application:cached
-      - ./docker/dev/nginx/nginx.conf:/etc/nginx/conf.d/default.conf:ro
+    deploy:
+      resources:
+        limits:
+          cpus: '0.25'
+          memory: 128M
+    environment:
+      - REDIS_HOST=redis
     ports:
       - 8001:80
+    volumes:
+      - .:/application:cached
+    working_dir: /application
 
   phpfpm:
     build:
-      context: docker/dev/php-fpm
+      context: .
+      dockerfile: sys/docker/alpine-phpfpm/Dockerfile
       network: host
     container_name: badge-poser-phpfpm
-    working_dir: /application
     depends_on:
       - redis
-    volumes:
-      - .:/application:cached
-      - ./docker/dev/php-fpm/php-ini-overrides.ini:/etc/php/7.4/fpm/conf.d/99-overrides.ini:ro
+    deploy:
+      resources:
+        limits:
+          cpus: '0.5'
+          memory: 1024M
+    env_file:
+      - ./.env
     ports:
-      - 9000
-
-  node:
-    container_name: badge-poser-node
-    image: node:12-alpine
-    user: "node"
+      - 9000:9000
     volumes:
       - .:/application:cached
+      - ./sys/php/php.ini-dev:/usr/local/etc/php/php.ini:ro
     working_dir: /application
 
   redis:
-    image: redis:alpine
     container_name: badge-poser-redis
+    image: redis:6.0.8-alpine3.12
     ports:
       - 6379:6379
+    sysctls:
+      net.core.somaxconn: 511
```

Comment:
it's to suppress a warning that it's raised by redis when running on docker-compose.

Reply:
```
# TCP listen() backlog.
#
# In high requests-per-second environments you need a high backlog in order
# to avoid slow clients connection issues. Note that the Linux kernel
# will silently truncate it to the value of /proc/sys/net/core/somaxconn so
# make sure to raise both the value of somaxconn and tcp_max_syn_backlog
# in order to get the desired effect.
tcp-backlog 511
```
Ref: https://github.com/redis/redis/blob/6f3b91fe26990f4c849d3f20269d69f3e3dc96f5/redis.conf#L100

it's just for dev purposes since in prod we use elasticache (and I believe those settings are totally different).

Reply:
the reference you provide is stating to use that value for backlog .. no mention of the value to be used to "be sure" for somaxconn AFAICS ... so it only suggest to always have somaxconn >= tcp-backlog ... 


my point is not to be so strict ....  somaxconn is limiting the total system amount of conn .. if you have 511 allocated for tcp backlog you don't have any for other stuff ... 1024 its the minimum reasonable IMHO

Reply:
done